### PR TITLE
UI/UX Enhancements Across Multiple Screens

### LINKED ISSUE #150: overall Ui fixes

### PR DESCRIPTION
**Pull request by eleonoraka**
## Summary

This PR introduces various UI improvements and consistency fixes across several screens to enhance user experience and visual coherence. Adds also a new feature, image cropping in AddTripScreen and EditProfileScreen when choosing a photo from the galley

## Changes

Overview Screen
- Implemented consistent color schemes for light/dark themes
- Made trip cards responsive with adaptable sizing
- Standardized date format display

Edit Profile Screen
- Removed redundant "Add" button
- Added placeholder text: "Press Enter to add interest"
- Enlarged interest input bar to match other field sizes

Add Trip Screen
- Added image cropping functionality for trip photos
- Fixed image selection button placement
- Improved overall layout consistency

Search User Screen
- Implemented light/dark theme color schemes
- Added text truncation with ellipsis for long usernames
- Standardized Add button size
- Fixed search bar behavior to prevent unwanted line breaks on Enter press

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #150 , #160 , #161
- [x] Tests have been added for new functionality.
- [ ] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing

Explain how this feature was tested and what kind of testing (unit, integration, UI) has been done. Mention platforms or configurations used for testing.
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

If you haven't added tests, please create a new issue and assign it.

##  Related Issues/Tickets

Closes #150 #160 #161

## Remarks

The file ImageCropperUtil doesn't reach the needed code coverage since it instantiates final class which are not mocked by mockito

### CODE REVIEW

**Comment by mirceagavrizi** [comment]
Thank you for the changes, already looking much better!
However, there are still some small changes to be done:
- in overview, center the trip cards
- in the settings screen and add trip screen the "Select Image from Gallery" button is not placed correctly
- in the edit profile screen, the interests also should be aligned from center + make it scrollable

Although, I suggest you merge these changes and create another PR for the rest.

**Comment by 1andreea26** [comment]
Please fix the conflicting files with the main.

File: app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt [hunk 1]
```diff
@@ -315,6 +315,31 @@ class AddTripScreenTest {
...
+    // Verify the image container has correct aspect ratio modifier
+    composeTestRule.onNodeWithTag("imageContainer").assertExists()
+  }
+
+  @Test
+  fun addTripScreen_imageHeight_respectsScreenSize() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    // Verify image height is calculated correctly based on screen width
+    composeTestRule.onNodeWithTag("imageContainer").assertExists()
```
Comment (mirceagavrizi) [inline app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt:331]: This test is the same as the one above

File: app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt [hunk 2]
```diff
@@ -350,7 +339,81 @@ class AddTripScreenTest {
...
+    // Mock the permission check to return PERMISSION_GRANTED
+    `when`(ContextCompat.checkSelfPermission(context, "android.permission.READ_MEDIA_IMAGES"))
+        .thenReturn(PackageManager.PERMISSION_GRANTED)
+
+    val result = checkFullPermission(context)
+    assertTrue(result) // Expecting true because permission is granted
+  }*/
+
+  /*@Test
+  fun checkPermissionReturnsTrueLimitedPermission() {
```
Comment (1andreea26) [inline app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt]: Are these tests not passing anymore, or why are they commented?

File: app/src/main/java/com/android/voyageur/ui/profile/EditProfileScreen.kt [hunk 3]
```diff
@@ -189,8 +165,8 @@ fun EditProfileScreen(userViewModel: UserViewModel, navigationActions: Navigatio
                               value = newInterest,
                               onValueChange = { newInterest = it },
                               label = { Text("Add Interest") },
-                              maxLines = 1,
```
Comment (mirceagavrizi) [inline app/src/main/java/com/android/voyageur/ui/profile/EditProfileScreen.kt:192]: Here leave maxLines = 1, as it is enough for writing an interest

File: app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt (hunk 2 above)
Comment (eleonoraka) [reply app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt]: I merged what I found when trying to merge with main. I don't know why they're commented

**Comment by 1andreea26** [comment]
The image cropper is a nice addition, however I would've put it in a different PR since this one is a bit long! Just a suggestion for the future :)

File: app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt (hunk 2 above)
Comment (1andreea26) [reply app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt]: It's my bad, I think I left them commented when I moved them to PermissionButtonGalleryTest and they appeared as changes in your PR. I'm sorry for this, could you please delete them? Thanks!

**Comment by 1andreea26** [comment]
It would've also been nice to have a screenshot of the image cropper attached to the PR, also as a suggestion for the future!

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt [hunk 4]
```diff
@@ -135,119 +142,142 @@ fun TripItem(tripsViewModel: TripsViewModel, trip: Trip, navigationActions: Navi
...
 
 @SuppressLint("StateFlowValueCalledInComposition")
 @Composable
-fun DisplayParticipants(trip: Trip) {
-  val configuration = LocalConfiguration.current
-  val screenWidth = configuration.screenWidthDp.dp
-  val avatarSize = (screenWidth * 0.06f).coerceIn(24.dp, 32.dp)
-  val numberOfParticipants = trip.participants.size
+fun DisplayParticipants(trip: Trip, userViewModel: UserViewModel) {
+  val numberOfParticipants = trip.participants.size - 1
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:204]: Why are you subtracting 1 here?

**Comment by eleonoraka** [comment]
> The image cropper is a nice addition, thanks for implementing it! However I would've put it in a different PR since this one is a bit long! Just a suggestion for the future :)

Thank you! I agree with you, I should have put it in another PR. I'll keep it in mind for the times! :)

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt [hunk 5]
```diff
@@ -135,119 +142,142 @@ fun TripItem(tripsViewModel: TripsViewModel, trip: Trip, navigationActions: Navi
...
   val numberToString = generateParticipantString(numberOfParticipants)
-
+  val themeColor = MaterialTheme.colorScheme.onSurface
   Column(
       modifier = Modifier.fillMaxHeight().padding(start = 0.dp, end = 0.dp, top = 8.dp),
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:208]: Are the start = 0.dp and end = 0.dp necessary here?

File: app/src/main/java/com/android/voyageur/ui/utils/ImageCropperUtil.kt [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+fun rememberImageCropper(
+    aspectRatioX: Int = 16,
+    aspectRatioY: Int = 9,
+    onResult: (ImageCropperResult) -> Unit
+): (Uri?) -> Unit {
+  val context = LocalContext.current
+
+  // Initialize the image cropper launcher
+  val cropImage =
+      rememberLauncherForActivityResult(CropImageContract()) { result ->
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/utils/ImageCropperUtil.kt:73]: Here it seems like I have some errors on your branch (It is up to date). I attached a screenshot below. Do you also have these errors on your machine?
Screenshot 2024-11-14 at 17 23 05

Comment (eleonoraka) [reply app/src/main/java/com/android/voyageur/ui/utils/ImageCropperUtil.kt:73]: On my machine everything works. It is possible that some imports have not been updated, sometimes it happens to me. Have you tried to update the imports? Let me know if the problem persists!

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt (hunk 4 above)
Comment (eleonoraka) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt]: I was having some issues with the OverviewScreen so I decided to use the one in the main branch. These changes were already there.

**Review by mirceagavrizi — APPROVED** [review]
Thank you for making the changes!

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt (hunk 4 above)
Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt]: I see, I don't know why they appear as your changes, but Mihai subtracted 1 for the user to not count as a participant to the trip. :)

File: app/src/main/java/com/android/voyageur/ui/utils/ImageCropperUtil.kt (hunk 6 above)
Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/utils/ImageCropperUtil.kt:73]: Fixed, it was a gradle problem. Thanks!

**Pull request merged**